react/features/base/conference/middleware.any.ts

```diff
@@ -195,64 +198,68 @@ function _conferenceFailed({ dispatch, getState }: IStore, next: Function, actio
         break;
     }
     case JitsiConferenceErrors.CONFERENCE_MAX_USERS: {
-        if (typeof APP === 'undefined') {
-            // in case of max users(it can be from a visitor node), let's restore
-            // oldConfig if any as we will be back to the main prosody
-            const newConfig = restoreConferenceOptions(getState);
-
-            if (newConfig) {
-                dispatch(overwriteConfig(newConfig)) // @ts-ignore
-                    .then(dispatch(conferenceWillLeave(conference)))
-                    .then(conference.leave())
-                    .then(dispatch(disconnect()))
-                    .then(dispatch(connect()));
-            }
+        dispatch(showErrorNotification({
+            hideErrorSupportLink: true,
+            descriptionKey: 'dialog.maxUsersLimitReached',
+            titleKey: 'dialog.maxUsersLimitReachedTitle'
+        }, NOTIFICATION_TIMEOUT_TYPE.LONG));
+
+        // In case of max users(it can be from a visitor node), let's restore
+        // oldConfig if any as we will be back to the main prosody.
+        const newConfig = restoreConferenceOptions(getState);
+
+        if (newConfig) {
+            dispatch(overwriteConfig(newConfig)) // @ts-ignore
+                .then(dispatch(conferenceWillLeave(conference)))
+                .then(conference.leave())
+                .then(dispatch(disconnect()))
+                .then(dispatch(connect()));
         }
 
         break;
     }
     case JitsiConferenceErrors.OFFER_ANSWER_FAILED:
         sendAnalytics(createOfferAnswerFailedEvent());
         break;
+
     case JitsiConferenceErrors.REDIRECTED: {
-        // once conference.js is gone this can be removed and both
-        // redirect logics to be merged
-        if (typeof APP === 'undefined') {
-            const newConfig = getVisitorOptions(getState, error.params);
-
-            if (!newConfig) {
-                logger.warn('Not redirected missing params');
-                break;
-            }
+        const newConfig = getVisitorOptions(getState, error.params);
 
-            const [ vnode ] = error.params;
+        if (!newConfig) {
+            logger.warn('Not redirected missing params');
+            break;
+        }
 
-            dispatch(overwriteConfig(newConfig)) // @ts-ignore
-                .then(dispatch(conferenceWillLeave(conference)))
-                .then(conference.leave())
-                .then(dispatch(disconnect()))
-                .then(dispatch(setIAmVisitor(Boolean(vnode))))
+        const [ vnode ] = error.params;
 
-                // we do not clear local tracks on error, so we need to manually clear them
-                .then(dispatch(destroyLocalTracks()))
-                .then(dispatch(connect()));
-        }
+        dispatch(overwriteConfig(newConfig)) // @ts-ignore
+            .then(dispatch(conferenceWillLeave(conference)))
+            .then(conference.leave())
+            .then(dispatch(disconnect()))
```

Comment:
It looks like the `disconnect` action will also leave the conference and `dispatch` some more actions. Why do we need to first manually leave the conference?

Also it seems that when we were calling `disconnect` from conference.js it was also sending IFrame API event. We don't need this anymore?

Reply:
Not sure about this disconnect event ... as this is visitors and is a new thing ... we need to decide how will that look from iframeAPI perspective in the future.